dice_ml/data_interfaces/base_data_interface.py

```diff
@@ -22,6 +22,10 @@ def _validate_and_set_outcome_name(self, params):
         else:
             raise ValueError("should provide the name of outcome feature as a string")
 
+    def _validate_outcome_as_last_column(self):
```

Comment:
Why is the outcome column need to be the last column in the dataframe?

Reply:
Let's say that the outcome (O) is the second column (column #2), and there are 4 inputs - A, B, C, D (all continuous for eg)
The dataset columns are like this: A, O, B, C, D
When initializing the data interface, the continuous column indices are (0, 2, 3, 4)

Now, when we give the query_instance as X_train, the columns would be like this: A, B, C, D

Now, we normalize the query instance before computing the proximity_loss. 
In this statement -- `for feature_index in self.continuous_feature_indexes:` there is an error because the continuous feature indices are (0, 2, 3, 4), and when we reach index 4, there is an error as the feature_index range in the query_instance is (0, 1, 2, 3)

Reply:
I wonder if we should fix the code to ensure that it can handle outcome column in any order. But then we may lose the efficiency of simply accessing the features by index values.

Reply:
Yup, it can be fixed by referring to each feature by the name and not the index

Reply:
@soundarya98 I feel that may be a better solution to avoid any future indexing bugs. Do you mind updating this PR by referring to each feature by name?
@gaugup  what do you think?

Reply:
Sure, I'll update the PR with the change!